### PR TITLE
CHET-159: Manage s3 sync in target

### PR DESCRIPTION
**Pull request by bpar476**
# Summary

* Implement running of status script on target host
* Implement parsing of status script output
* Integrate running and waiting for download with file system migration

### CODE REVIEW

**Comment by bpar476** [comment]
This is pending bean definnitions

**Comment by bpar476** [comment]
This also does not include surfacing the real-time status of the s3 download to the user via REST endpoint. I'm not sure what the best approach for this is. Let's track it separately and accomplish it when we do tackle the UI.

File: src/main/java/com/atlassian/migration/datacenter/core/fs/download/s3sync/S3SyncFileSystemDownloader.java [hunk 1]
```diff
@@ -50,28 +51,40 @@ public void initiateFileSystemDownload() throws CannotLaunchCommandException {
...
+            logger.error("error launching s3 sync command", e);
+            throw new CannotLaunchCommandException("unable to launch file system download command successfully.");
+        } catch (SuccessfulSSMCommandConsumer.SSMCommandInvocationProcessingError never) {
+        }
+    }
 
-            if (status.equals(CommandInvocationStatus.SUCCESS)) {
-                return;
-            }
+    public S3SyncCommandStatus getFileSystemDownloadStatus() throws IndeterminateS3SyncStatusException {
```
Comment (nanux) [inline src/main/java/com/atlassian/migration/datacenter/core/fs/download/s3sync/S3SyncFileSystemDownloader.java]: `@Nullable` or should we provide javadoc when `null` can be returned?

File: src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java [hunk 2]
```diff
@@ -140,15 +150,29 @@ public void startMigration() throws InvalidMigrationStageError {
...
         try {
             fsUploader.uploadDirectory(getSharedHomeDir());
         } catch (FileUploadException e) {
             logger.error("Caught exception during upload; check report for details.", e);
         }
 
+        if (!report.getStatus().equals(FAILED)) {
+            logger.trace("upload of shared home complete. commencing shared home download");
+            try {
+                fileSystemDownloadManager.downloadFileSystem();
```
[the remark this answers is not in the document]
Comment (bpar476) [reply src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java]: Do you mean catching the exception and setting the status?

**Comment by bpar476** [comment]
> Looks good, there might be some minor improvements but I would suggest to not add more non-critical changes to the PR and get it to `master`.

Got it :D I've created this to do the cleanup you've listed
https://github.com/atlassian-labs/dc-migration-assistant/pull/72#pullrequestreview-377560245

**Pull request merged**